src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -736,23 +737,23 @@ char* java_lang_String::as_utf8_string(oop java_string, typeArrayOop value, int
   }
 }
 
-bool java_lang_String::equals(oop java_string, const jchar* chars, int len) {
+bool java_lang_String::equals(oop java_string, const jchar* chars, int num_unicode_points) {
```

Comment:
Please undo these changes - these are not "unicode points".

Reply:
I don't think that you're correct here, David. As far as I understand, the length `len` does not represent the length of the `chars` array but the number of unicode code points in the array. The same is true for the other overloads, such as the UTF8 string. I'd appreciate it if you could explain why what I'm saying is incorrect here.

Reply:
The `len` parameter is just the length of the array. We use it to iterate through the array. e.g.
```
    for (int i = 0; i < len; i++) {
      if (value->char_at(i) != chars[i]) {
        return false;
      }
    }
```
But you are right that this is also the number of "code points" as defined by the Unicode standard - my apologies for that. I was mistakenly thinking that codepoints refer to actual characters, but it is a general term used for any numeric value in the unicode codespace, including non-characters and importantly surrogates. But while technically accurate I don't think using `num_unicode_points` adds any value here - quite the opposite as it obscures that this is simply the length of the array.

Reply:
>The len parameter is just the length of the array. We use it to iterate through the array. e.g.

Reply:
I think we need to be very clear when we are passing in the expected length of an array as opposed to the number of characters/elements/points we intend to try to read from it. To that end just as we often use `void foo(char* buf, int buflen)` we should use e.g. `(char* utf8_str, int utf8_str_len)` if that is what we actually mean. (Though note we have an issue with using int instead of size_t for length!).